CMakeLists.txt

```diff
@@ -13,13 +13,19 @@ if(CMAKE_BUILD_TYPE MATCHES "Release")
 	endif()
 endif()
 
+set(_gnulike GNU Clang)
+
 if(MSVC)
 	set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
 	string(APPEND CMAKE_CXX_FLAGS " /W4 /GR- /Zc:__cplusplus")
-else()
+elseif(CMAKE_CXX_COMPILER_ID IN_LIST _gnulike)
 	string(APPEND CMAKE_CXX_FLAGS " -Wno-deprecated -fdiagnostics-color")
 endif()
 
+if(NOT MSVC)  # MSVC is too strict right now for Ninja code with C++17
+	set(CMAKE_CXX_STANDARD 17)
```

Comment:
IIRC the Modern CMake way to do this is `target_compile_features` with `cxx_std_17`.

I also wonders what happens on compilers that don't support C++17?

Reply:
Yes that would be better way to do it. 

CMake by default is supposed to gracefully degrade for older compilers. There are naturally corner cases not anticipated. Like the option above, it may be safe to enable this only when known to be needed -- e.g. for Intel icl/icpc

Reply:
It turns out that target_compile_features() breaks older compilers like GCC 4.8 since that function strictly requires C++17, while the code supports graceful fallback and so instead we use 8cef6933d1dbd
[CXX_STANDARD](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html) property which also decays gracefully.

Reply:
...however, there are still remaining bugs invoked by *carte blanche* setting of C++17 standard due to `using namespace std` in Ninja code. So in the end a compiler-specific if(Intel) was the least-change way of doing this.

Reply:
> IIRC the Modern CMake way to do this is target_compile_features with cxx_std_17.

SET(CMAKE_CXX_STANDARD 17) if you want it to apply to all target in the project.

Reply:
> carte blanche setting of C++17 standard due to using namespace std in Ninja code.

@scivision 

Does this PR resolve that problem? Or would it also need to remove "using namespace std" in implementation files as well ? https://github.com/ninja-build/ninja/pull/1614

Reply:
I tried #1614 with MSVC 16.5.2 and the error still existed:

```
error C2872: 'byte': ambiguous symbol
```

Reply:
Ah, ok, thanks for the information.